core/main/src/service/apps/apps_updater.rs

```diff
@@ -126,18 +150,304 @@ pub async fn update(mut client: ExtnClient, apps_update: AppsUpdate, ignore_list
         installed_apps.clone()
     );
 
-    for app in apps_update.apps {
-        let installed_ids = installed_apps.clone();
-        if !installed_ids.into_iter().any(|installed_app| {
-            installed_app.id.eq(&app.id) && installed_app.version.eq(&app.version)
-        }) {
-            debug!("update: Application is not currently installed, installing: title={}, id={}, version={}", app.title, app.id, app.version);
-            let metadata = AppMetadata::new(app.id, app.title, app.version, app.uri, app.data);
-            let resp = client.request(AppsRequest::InstallApp(metadata)).await;
-
-            if let Err(e) = resp {
-                error!("update: Could not install app: e={:?}", e);
+    let mut app_list: Vec<app_catalog::AppMetadata> = apps_update
```

Comment:
Refactored this using `filter` collection operation. This is so we can filter out the already installed apps before sorting.